src/UseCase/BannerSelection/CampaignConfigurationLoader.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\BannerServer\UseCase\BannerSelection;
+
+use FileFetcher\FileFetcher;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\Yaml\Yaml;
+
+/**
+ * @license GNU GPL v2+
+ */
+class CampaignConfigurationLoader {
+
+	private const SEASONAL_IMPRESSION_LIMIT = 10;
+
+	private $fileSystem;
+	private $fileFetcher;
+
+	public function __construct( Filesystem $fileSystem, FileFetcher $fileFetcher ) {
+		$this->fileSystem = $fileSystem;
+		$this->fileFetcher = $fileFetcher;
+	}
+
+	private function loadFiles( string ...$configFiles ): array {
+		$configs = [];
+		foreach ( $configFiles as $file ) {
+			if ( $this->fileSystem->exists( $file ) ) {
```

Comment:
Haha, it's just your code from FundraisingFrontend :smile:. I was thinking of implementing something similar what you did there and just make some finer adjustments and this PR was more of a reminder that this is still completely missing from the code and was not really accounted for in our planning. 

I will see if I can get to this today still, I am currently still working on the use case.

Reply:
I know, I recognized the code from the CampaignConfigurationLoader and remembered the problems we had with it - mostly the "why is my config not working?" because you can specify the config files in the application config and if there is at least 1 one, it will try to load it, leading to a different config you expected. 

I sometimes have this tendency to make code more "robust" by adding error recovery, but in most cases the recovery just masks the errors and makes them harder to debug. The `Filesystem` code only makes sense when you have a candiate list of files and want to pick the 1st one